### DISCUSSION TITLE
esbuild postinstall version mismatch error when deploying to vercel

### DISCUSSION THREAD

**Discussion by sbkl** [vercel/community#4361] [Help] [answered]
### Summary

Context monorepo using turborepo with pnpm. When deploying to vercel, I get a esbuild version mismatch error with the following log. When I check the node_modules at the root of the monorepo, esbuild package.json version is 0.17.11 but not sure from which package this is coming from. How to fix this version mismatch?

```
../../node_modules/esbuild postinstall: /vercel/path0/node_modules/esbuild/install.js:94
--
00:04:24.960 | ../../node_modules/esbuild postinstall:     throw new Error(`Expected ${JSON.stringify("0.16.4")} but got ${JSON.stringify(stdout)}`);
00:04:24.961 | ../../node_modules/esbuild postinstall:           ^
00:04:24.961 | ../../node_modules/esbuild postinstall: Error: Expected "0.16.4" but got "0.17.11"
00:04:24.961 | ../../node_modules/esbuild postinstall:     at validateBinaryVersion (/vercel/path0/node_modules/esbuild/install.js:94:11)
00:04:24.961 | ../../node_modules/esbuild postinstall:     at /vercel/path0/node_modules/esbuild/install.js:243:5
```
Recently 

### Steps to Reproduce

Just pushing my code to github which deploys to vercel.

**Comment by amyegan** [comment, marked as answer]
If you remove `node_modules` and try to install packages again in your local environment, do you get the same error?

**Reply by sbkl** [reply]
Super weird one. So followed your advised, downloaded the latest commit from my github repo to start fresh and then I couldnt pnpm install anymore even locally.
I had a look to my pnpm-lock.yaml file and identified the inconsistency below where the esbuild version was 0.17.1 but all dependencies 0.16.4.
So I tried my luck, changed manually the `/esbuild@0.17.1` to `/esbuild@0.16.4` and ran pnpm install again. It complained about my nextjs app couldn't find the preferred version of esbuild and went to fix it. Then deployed and it went through.
No idea what I have done but at least now I can deploy. Something I didn't mentioned is that I was upgrading from next 13.1 to 13.5.4. Maybe something messed up in the process.

```
  /esbuild@0.17.1:
    resolution: {integrity: sha512-qQrPMQpPTWf8jHugLWHoGqZjApyx3OEm76dlTXobHwh/EBbavbRdjXdYi/GWr43GyN0sfpap14GPkb05NH3ROA==}
    engines: {node: '>=12'}
    hasBin: true
    requiresBuild: true
    optionalDependencies:
      '@esbuild/android-arm': 0.16.4
      '@esbuild/android-arm64': 0.16.4
      '@esbuild/android-x64': 0.16.4
      '@esbuild/darwin-arm64': 0.16.4
      '@esbuild/darwin-x64': 0.16.4
      '@esbuild/freebsd-arm64': 0.16.4
      '@esbuild/freebsd-x64': 0.16.4
      '@esbuild/linux-arm': 0.16.4
      '@esbuild/linux-arm64': 0.16.4
      '@esbuild/linux-ia32': 0.16.4
      '@esbuild/linux-loong64': 0.16.4
      '@esbuild/linux-mips64el': 0.16.4
      '@esbuild/linux-ppc64': 0.16.4
      '@esbuild/linux-riscv64': 0.16.4
      '@esbuild/linux-s390x': 0.16.4
      '@esbuild/linux-x64': 0.16.4
      '@esbuild/netbsd-x64': 0.16.4
      '@esbuild/openbsd-x64': 0.16.4
      '@esbuild/sunos-x64': 0.16.4
      '@esbuild/win32-arm64': 0.16.4
      '@esbuild/win32-ia32': 0.16.4
      '@esbuild/win32-x64': 0.16.4
    dev: false
```
